foundations/javascript_basics/fundamentals-5.md

```diff
@@ -28,47 +28,116 @@ As an example of what we mean, let's consider a `sumOfTripledEvens` function. It
 - For every even number, it will triple it
 - Then it will sum all those even numbers
 
-Currently, if you wanted to implement a function like that you may write something similar to this:
+Can you try thinking of how you could implement a function like that using pseudocode?
+  
+1. We need to perform an operation only on the even numbers.
+1. We need to transform *those* numbers into triple numbers.
+1. Finally, we need to add the result up from the previous transformation.
+
+So using that logic, you may end up implementing something like this:
 
 ```javascript
 function sumOfTripledEvens(array) {
   let sum = 0;
   for (let i = 0; i < array.length; i++) {
+    // Step 1: If the element is an even number
     if (array[i] % 2 === 0) {
-      sum += array[i] * 3;
+      // Step 2: Multiply this number by three
+      const tripleEvenNumber = array[i] * 3;
+
+      // Step 3: Add the new number to the total
+      sum += tripleEvenNumber;
     }
   }
   return sum;
 }
 ```
 
-We have a function `sumOfTripledEvens` that takes in an `array`, it then iterates through the array and for every even number it adds triple that number to sum. After the `for` loop is complete, it returns the sum.
+In the above code, there are 3 important snippets to consider:
 
-So that works, but there are more elegant ways of going about this. For instance, you could re-write the above as follows:
+- `if (array[i] % 2 === 0)`: checks if a given number is even.
+- `array[i] * 3;`: multiply this number by three.
+- `sum += tripleEvenNumber`: add that number to the total.
 
-```javascript
-function sumOfTripledEvens(array) {
-  return array
-    .filter(x => x % 2 === 0)
-    .map(x => x * 3)
-    .reduce((accumulator, current) => accumulator + current, 0);
+Every single piece solves a cruicial problem with our code.
+But it is possible to make the function more concise and readable by using some built-in array methods. 
+These methods are slightly more complicated than you've been used to, so let's take a moment to understand how to use them.
+
+#### The .map() method
+
+`.map(callback)` is one such function. It expects a `callback` as an argument, which is a fancy way to say "I want to pass another function as an argument to my function".
+
+Let's say we had a function `addOne(num)`, which takes in `num` as an argument and outputs that `num` increased by 1. 
+And let's say we had an array of numbers, `const arr = [1, 2, 3, 4, 5]` and we'd like to increase all of these numbers by `+1` using our `addOne(num)` function.
+Instead of making a `for` loop and iterating over the above array, we could use our `.map(callback)` array method instead, which **automatically** iterates over an array for us. 
```

Comment:
```suggestion
Instead of making a `for` loop and iterating over the above array, we could use our `map` array method instead, which **automatically** iterates over an array for us. 
```